test/mdset.c

```diff
@@ -0,0 +1,714 @@
+/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+ * Copyright by The HDF Group.                                               *
+ * Copyright by the Board of Trustees of the University of Illinois.         *
```

Comment:
If this is new code, we probably want to avoid the University of Illinois portion here. Otherwise, it's unclear whether/how that portion applies.